### PR TITLE
Application of electron residual corrections

### LINKED ISSUE #16: E/gamma energy corrections implementation missing
For e/gamma there is no implementation for [residual scale and smearings corrections](https://twiki.cern.ch/twiki/bin/view/CMS/EgammaMiniAODV2#Energy_Scale_and_Smearing) available. This should be implemented. Naturally, I recommend to do this in the placeholder function [`AdditionalCorrections`](https://github.com/KIT-CMS/Artus/blob/master/KappaAnalysis/src/Producers/ElectronCorrectionsProducer.cc#L84) in the `ElectronCorrectionsProducer`.

### PR DESCRIPTION
**Pull request by HerrHorizontal**
By calculating correction factors from the UserFloats stored in the electronID object added to the KappaEvent, correction factors get calculated and applied on the electron p4.
The UserFloats get efficiently accessed by the corresponding index, which is identified by a single string matching on reading the lumi block information.

### CODE REVIEW

**Comment by HerrHorizontal** [comment]
After the last correction the application of correction factors works fine. Request to merge.

File: KappaAnalysis/src/Producers/ElectronCorrectionsProducer.cc [hunk 1]
```diff
@@ -84,5 +106,11 @@ void ElectronCorrectionsProducer::Produce(KappaEvent const& event, KappaProduct&
 void ElectronCorrectionsProducer::AdditionalCorrections(KElectron* electron, KappaEvent const& event,
                                                         KappaProduct& product, KappaSettings const& settings) const
 {
+	//size_t iter = 0;
```
Comment (mschnepf) [inline KappaAnalysis/src/Producers/ElectronCorrectionsProducer.cc]: If `iter` is no longer needed, you can delete this line.

Comment (HerrHorizontal) [reply KappaAnalysis/src/Producers/ElectronCorrectionsProducer.cc]: Done

Comment (mschnepf) [reply KappaAnalysis/src/Producers/ElectronCorrectionsProducer.cc]: Nice, thanks.

**Pull request merged**